src/extension_commands.ml

```diff
@@ -131,6 +133,124 @@ let _open_current_dune_file =
   in
   command Extension_consts.Commands.open_current_dune_file handler
 
+let _jump_to_prev_hole, _jump_to_next_hole =
+  let pick_next_hole current_pos ~sorted_non_empty_holes_list:holes =
+    (* we want to find such a range that starts after the current position *)
+    let next_hole =
+      List.find holes ~f:(fun range ->
+          match Position.compare current_pos (Range.start range) with
+          | Ordering.Less -> true
+          | Greater -> false
+          | Equal ->
+            (* we don't want the same range that we're in now; we need the next
+               one *)
+            not (Range.contains range ~positionOrRange:(`Position current_pos)))
+    in
+    (* - if the current position is larger than all other ranges, we cycle back
+       to first hole in the file
+
+       - [hd_exn] because [holes] is non-empty *)
+    Option.value_or next_hole ~default:(fun () -> List.hd_exn holes)
```

Comment:
You could avoid the comment and the use of `hd_exn` by special casing the one element list:

```ocaml
let pick_next_hole current_pos ~sorted_non_empty_holes:holes =
  match holes with
  | [] -> assert false
  | [x] -> x
  | default :: _ -> let next_hole = ... in Option.value ~default holes
```

Reply:
I think you misunderstood what the code is trying to do (or I misunderstood what you mean). We traverse all holes looking a hole such that its range comes after the current position. If there is no such hole, then we pick the first hole in the file, assuming that all holes came before the current position, ie cycle back to the first hole. Does this make sense? 

A similar algo used in `pick_prev_hole`.

Reply:
> Does this make sense?

It makes sense. I think the snippet above has this exact behavior. The full traversal of holes would still occur in the `...` of `let next_hole = ...`.